test/integration/tags/standard_tag_test.rb

```diff
@@ -31,9 +31,10 @@ def test_has_a_block_which_does_nothing
 
     assert_template_result('', '{%comment%}{%blabla%}{%endcomment%}')
     assert_template_result('', '{% comment %}{% blabla %}{% endcomment %}')
+    assert_template_result('', '{% comment %}{% if "foo" %}{% endcomment %}')
     assert_template_result('', '{%comment%}{% endif %}{%endcomment%}')
     assert_template_result('', '{% comment %}{% endwhatever %}{% endcomment %}')
-    assert_template_result('', '{% comment %}{% raw %} {{%%%%}}  }} { {% endcomment %} {% comment {% endraw %} {% endcomment %}')
+    assert_template_result(' ', '{% comment %}{% raw %} {{%%%%}}  }} { {% endcomment %} {% comment {% endraw %} {% endcomment %}')
```

Comment:
For reasons I have yet to discover, `CommentBody` parses this as

1. `"{% comment %}{% raw %} {{%%%%}}  }} { {% endcomment %}"`
2. `" "`
3. `"{% comment {% endraw %} {% endcomment %}"`

whereas `BlockBody` treats the whole thing as a single comment.

Reply:
Found it. For `BlockBody`, `{% raw %}` swallows up the first `{% endcomment %}`.
```
"{% comment %}"
    "{% raw %}"
        " {{%%%%}}  }} { {% endcomment %} {% comment "
    "{% endraw %}"
    " "
"{% endcomment %}"
```

Reply:
I feel like this workaround (raw in comment) could be pretty common and that this change will subtly break some uses of `{% comment %}` :(

Reply:
cc @fw42 ^ if this test breaks how bad would it be?